core/dataset.cc

```diff
@@ -944,38 +945,159 @@ generateRowsWhere(const SqlBindingScope & scope,
             return dynamic_cast<const ArithmeticExpression *>(&expression);
         };
 
+    auto getBoundParameter = [] (const SqlExpression & expression) -> const BoundParameterExpression *
+        {
+            return dynamic_cast<const BoundParameterExpression *>(&expression);
+        };
+
+    //look for rowName() != constant-or-bound 
+    //or constant-or-bound != rowName()
+    auto isRowNameFilter = [&](const SqlExpression & expression)
+    {
+        auto comparison = dynamic_cast<const ComparisonExpression *>(&expression);
+
+        if (comparison) {
+
+            auto clhs = getConstant(*comparison->lhs);
+            auto crhs = getConstant(*comparison->rhs);
+            auto flhs = getFunction(*comparison->lhs);
+            auto frhs = getFunction(*comparison->rhs);
+            auto blhs = getBoundParameter(*comparison->lhs);
+            auto brhs = getBoundParameter(*comparison->rhs);
+
+            if (frhs && (clhs || blhs) && comparison->op == "!=") {
```

Comment:
There is for the generic case. in the case of "blah AND rowName() == constant", you would probably want to do something different though.